### PR TITLE
Feature of adding product manifest generating and removing

### PR DESCRIPTION
**Pull request by yma88**

### CODE REVIEW

File: charon/pkgs/maven.py [hunk 1]
```diff
@@ -440,7 +450,12 @@ def handle_maven_del(
     )
     logger.info("Files deletion done\n")
 
-    # 4. Use changed GA to scan s3 for metadata refreshment
+    # 4. Delete related manifest from s3
+    logger.info("Start deleting manifest from s3")
+    s3_client.delete_manifest(prod_key, target, manifest_bucket_name)
+    logger.info("Manifest deletion is done\n")
```
Comment (ligangty) [inline charon/pkgs/maven.py:456]: I think this step can be skipped with a log if manifest_bucket_name is not specified.

File: charon/pkgs/npm.py [hunk 2]
```diff
@@ -90,6 +96,9 @@ def handle_npm_uploading(
     valid_dirs = __get_path_tree(valid_paths, target_dir)
 
     prefix_ = remove_prefix(prefix, "/")
+
+    manifest_name, manifest_full_path = write_manifest(valid_paths, target_dir, product)
```
Comment (ligangty) [inline charon/pkgs/npm.py:100]: I think this step can be skipped with a log if manifest_bucket_name is not specified.

File: charon/pkgs/npm.py [hunk 3]
```diff
@@ -102,6 +111,10 @@ def handle_npm_uploading(
     )
     logger.info("Files uploading done\n")
 
+    logger.info("Start uploading manifest to s3")
+    client.upload_manifest(manifest_name, manifest_full_path, target, manifest_bucket_name)
+    logger.info("Manifest uploading is done\n")
```
Comment (ligangty) [inline charon/pkgs/npm.py:116]: I think this step can be skipped with a log if manifest_bucket_name is not specified.

File: charon/pkgs/npm.py [hunk 4]
```diff
@@ -177,6 +196,10 @@ def handle_npm_del(
     )
     logger.info("Files deletion done\n")
 
+    logger.info("Start deleting manifest from s3")
+    client.delete_manifest(product, target, manifest_bucket_name)
+    logger.info("Manifest deletion is done\n")
```
Comment (ligangty) [inline charon/pkgs/npm.py:201]: I think this step can be skipped with a log if manifest_bucket_name is not specified.

File: charon/pkgs/maven.py (hunk 1 above)
Comment (yma88) [reply charon/pkgs/maven.py:461]: @ligangty The case was already covered inside the delete_manifest method, others are sorted based on the comment.

**Comment by ligangty** [comment]
> The tests are failed caused by downloading from repo 'appstream'. It seems that the mirrors can't access? @ligangty

@yma88 Seems it is caused by centos8 yum repo issues. I've corrected it to use centos7, please rebase and try again.

**Comment by yma88** [comment]
> > The tests are failed caused by downloading from repo 'appstream'. It seems that the mirrors can't access? @ligangty
> 
> @yma88 Seems it is caused by centos8 yum repo issues. I've corrected it to use centos7, please rebase and try again.

@ligangty It's passed, thanks!

**Review by ligangty — APPROVED** [review]
LGTM

**Pull request merged**